elasticsearch/connection/http_urllib3.py

```diff
@@ -152,6 +155,10 @@ def perform_request(self, method, url, params=None, body=None, timeout=None, ign
 
             request_headers = self.headers
             if headers:
+                if self.http_compress == True:
+                    headers.update(urllib3.make_headers(accept_encoding=True))
+                    headers.update({'Content-Encoding': 'gzip'})
```

Comment:
the headers should be modified in `__init__` so it only happens once at initializatin and not with every single request, see where `self.headers` are created. Also please use lowercase for `content-encoding` to be consistent.